doc/modules/pipeline.rst

```diff
@@ -213,9 +213,11 @@ FeatureUnion: composite feature spaces
 :class:`FeatureUnion` combines several transformer objects into a new
 transformer that combines their output. A :class:`FeatureUnion` takes
 a list of transformer objects. During fitting, each of these
-is fit to the data independently. For transforming data, the
-transformers are applied in parallel, and the sample vectors they output
-are concatenated end-to-end into larger vectors.
+is fit to the data independently. It can also be used to apply different
+transformations to each field of the data, producing a homogeneous feature
+matrix from a heterogeneous data source.
+The transformers are applied in parallel, and the feature matrices they output
+are concatenated side-by-side into a larger matrix.
```

Comment:
Since this PR adds ColumnTransformer, we can say here something like: for data organized in fields with heterogeneous types, see the related class `class:ColumnTransformer`.